### DISCUSSION TITLE
MagickReadSettings proper usage

### DISCUSSION THREAD

**Discussion by nevaran** [dlemstra/Magick.NET#812] [General]
I tried using this settings class to supposedly pre-resize an image, but it doesnt seem to be doing **anything**. Am I understanding the class wrong or perhaps using it wrong?
What I am trying is to reduce memory usage spike if large images are being loaded and then resized; if theres any tips on doing that it is greatly appreciated

```
var settings = new MagickReadSettings
                {
                    Width = Settings.ThumbnailRes
                };

                using MagickImage image = new MagickImage(path, settings);
```

**Comment by dlemstra** [comment]
The width will only be used for specific image formats. There is no such thing as "auto resize" when reading the image.

**Reply by nevaran** [reply]
I was hoping there was some sort of way the image could be read in parts that are being resized before the next chunk is loaded which would possibly reduce the memory spike by a good chunk

**Reply by dlemstra** [reply]
I don't even know how we could do that. And if I did I would have no clue how we could do that in all the formats that we support because they all store their pixels differently.

**Reply by nevaran** [reply]
I see; Well at any rate you have an awesome library. Keep it up and happy new year and whatnot ;)
